### PR TITLE
Robustifying maplab

### PR DESCRIPTION
**Pull request by gauthamm**
Add changes from July release and reduce certain fatal checks to error checks 
---

Before opening this pull request, please consider the following **checklist**:

 - For bug fixes:
   - Describe the bug or add a link to the maplab issue.
   - Ideally include instructions on how to reproduce it.
   - Describe your solution.
 - For new features:
   - Describe the new functionality and provide instructions on how to use it.
   - Add results (plots, pictures, videos).
   - Make sure this feature can be used on the existing maplab dataset, otherwise provide an example dataset.
   - Add a unit/integration/end-to-end test for this new feature.

### CODE REVIEW

File: algorithms/landmark-triangulation/src/pose-interpolator.cc [hunk 1]
```diff
@@ -167,10 +162,9 @@ void PoseInterpolator::computeRequestedPosesInRange(
       current_state.head<kStateOrientationBlockSize>();
   state_linearization_point_begin.p_M_I =
       current_state.segment<kPositionBlockSize>(kStatePositionOffset);
-  constexpr bool kEmitWarningIfValuesOverwritten = false;
   state_buffer->addValue(
       state_linearization_point_begin.timestamp,
-      state_linearization_point_begin, kEmitWarningIfValuesOverwritten);
+      state_linearization_point_begin);
```
Comment (hendrik-skubch) [inline algorithms/landmark-triangulation/src/pose-interpolator.cc:167]: so you want warnings?

File: common/maplab-common/include/maplab-common/temporal-buffer.h [hunk 2]
```diff
@@ -23,86 +22,127 @@ class TemporalBuffer {
...
+  typename BufferType::reverse_iterator rend() {
+    return values_.rend();
+  }
+
+  typename BufferType::const_reverse_iterator rend() const {
+    return values_.crend();
   }
 
  protected:
+  BufferType& getValues() {
```
Comment (hendrik-skubch) [inline common/maplab-common/include/maplab-common/temporal-buffer.h:137]: Why does this need to stay protected?
Also: I'd prefer getValues and editValues() as it's non-const counter part

File: common/maplab-common/include/maplab-common/temporal-buffer-inl.h [hunk 3]
```diff
@@ -25,59 +23,69 @@ TemporalBuffer<ValueType, AllocatorType>::TemporalBuffer(
 template <typename ValueType, typename AllocatorType>
 TemporalBuffer<ValueType, AllocatorType>::TemporalBuffer(
     const TemporalBuffer<ValueType, AllocatorType>& other) {
-  // Lock both mutexes without deadlock.
-  std::lock(mutex_, other.mutex_);
```
Comment (hendrik-skubch) [inline common/maplab-common/include/maplab-common/temporal-buffer-inl.h:29]: are you sure about removing this?

File: common/maplab-common/include/maplab-common/temporal-buffer.h (hunk 2 above)
Comment (gauthamm) [reply common/maplab-common/include/maplab-common/temporal-buffer.h:137]: Changes to these files are based on https://github.com/ethz-asl/maplab/compare/pre_release_public/march-2018...devel/mav

File: common/maplab-common/include/maplab-common/temporal-buffer-inl.h (hunk 3 above)
Comment (gauthamm) [reply common/maplab-common/include/maplab-common/temporal-buffer-inl.h:29]: they were based on https://github.com/ethz-asl/maplab/compare/pre_release_public/march-2018...devel/mav and there weren't any issues while testing. Basically the locks are now taken care on a higher layer.

File: algorithms/landmark-triangulation/src/pose-interpolator.cc (hunk 1 above)
Comment (gauthamm) [reply algorithms/landmark-triangulation/src/pose-interpolator.cc:167]: This is a result of the API changes in temporal buffer. In the current version, it replaces the values with the same timestamps without warning.

File: common/maplab-common/include/maplab-common/temporal-buffer-inl.h (hunk 3 above)
Comment (hendrik-skubch) [reply common/maplab-common/include/maplab-common/temporal-buffer-inl.h:29]: cool!

File: algorithms/landmark-triangulation/src/pose-interpolator.cc (hunk 1 above)
Comment (hendrik-skubch) [reply algorithms/landmark-triangulation/src/pose-interpolator.cc:167]: hoki doki

**Review by hendrik-skubch — APPROVED** [review]

**Pull request merged**